ext/intl/tests/dateformat_get_best_pattern.phpt

```diff
@@ -0,0 +1,36 @@
+--TEST--
+IntlDateFormatter::getBestPattern()
+--SKIPIF--
+<?php
+if (!extension_loaded('intl'))	die('skip intl extension not enabled'); ?>
+<?php if (version_compare(INTL_ICU_VERSION, '50.1.2', '<')) die('skip for ICU < 50.1.2'); ?>
```

Comment:
As an aside:
This line is included in almost all intl dateformat tests - there, it is used in this form:
```php
<?php if (version_compare(INTL_ICU_VERSION, '50.1.2') >=  0) die('skip for ICU < 50.1.2'); ?>
```
This, however, ends up skipping the test on my machine (`INTL_ICU_VERSION === '67.1'`). I assume `version_compare` is used incorrectly here? Should this be fixed for all other tests as well (and switched to what is the more readable form in my opinion)? Is this line still necessary (as PHP 7.4 requires ICU 50.1?)

Reply:
I think only the skip message is a bit deceptive: if fact, `die('skip for ICU < 50.1.2');` means that the test is `skip`ped because it's `for ICU < 50.1.2`. It also took quite some time for me to get accustomed to this.

If ICU 50.1 is required now then these tests are probably  not in use anymore 🤔

Reply:
And I don't think the `version_compare`  is necessary for this test.

Reply:
Ah yes, that explains the message.
I still dont understand why these tests are not required for newer ICU versions - I would expect basic tests like
https://github.com/php/php-src/blob/d4530f807a78d50c2f5fd66114632480e7651d31/ext/intl/tests/dateformat_format.phpt#L5
to still be in use for newer versions/after PHP 7.4.
(For an incomplete list of affected tests, this is the commit that startes skipping a bunch of them for ICU >= 51.2 (was later changes to 50.1): 4840b0a749454335218f6131892031333d66142d)

Reply:
Tests for other ICU versions are under _variant2, _variant3 etc.

Reply:
> If ICU 50.1 is required now then these tests are probably not in use anymore thinking

I believe the reason why these tests still exist is that our requirement is ICU 50.1, while this checks for ICU 50.1.2.

Though maybe we can bump up the requirements slightly, it looks like RHEL 7 nowadays ships with ICU 50.2 (according to https://rpms.remirepo.net/rpmphp/zoom.php?rpm=icu). @remicollet Would a 50.2 requirement be safe?

Reply:
The same RHEL version comes with PHP 5: https://rpms.remirepo.net/rpmphp/zoom.php?rpm=php so is RHEL even supposed to dictate what PHP 8 should do?